File: leetcode/cpp/830_Positions_of_Large_Groups.cpp
#include <string>
#include <vector>
using namespace std;
class Solution {
public:
    vector<vector<int>> largeGroupPositions(string s) {
        vector<vector<int>> ret;
        int l = 0;

        for (int r = 0; r < s.size(); ++r) {
            if (r == s.size() - 1 || s[r] != s[r + 1]) {
                // we are at end of consec chars
                if (r - l + 1 >= 3) ret.push_back({l, r});
                
                l = r + 1;
            }
        }

        return ret;
	}
};

File: leetcode/cpp/830_Positions_of_Large_Groups.py
class Solution:
    def largeGroupPositions(self, s: str) -> List[List[int]]:
        ret = []
        l = 0

        for r in range(len(s)):
            if r == len(s) - 1 or s[r] != s[r + 1]:

                if r - l + 1 >= 3: ret.append([l, r]);
                l = r + 1

        return ret
